### PR TITLE
add word counter program

### PR DESCRIPTION
**Pull request by ramrom**

### CODE REVIEW

File: wordcount.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  if err != nil { panic(err) }
+
+  // I discovered I can use ioutil.ReadFile
+  stats, _ := os.Stat(filename)
+  filesize := stats.Size()
+  //fmt.Println(reflect.TypeOf(stats))
+  fmt.Printf("file size: %v bytes\n", filesize)
+
+  filecontent := make([]byte, filesize)
+  count, err2 := file.Read(filecontent)
```
Comment (kelmerp) [inline wordcount.go]: you can keep re-using err

File: wordcount.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+// Optimization 1: used a fixed sized byte buffer to stream parts of a file in at a time
+// Optimization 2: don't convert the file content to an array of words, directly insert into the map of string to int
+func main() {
+  filename := os.Args[1]
+  fmt.Printf("%v given as filename arg\n", filename)
+
+  file, err := os.Open(filename)
+  defer file.Close()
+  if err != nil { panic(err) }
```
Comment (kelmerp) [inline wordcount.go]: I know it looks nice, but we generally don't have the opening and closing brackets on one line
https://golang.org/doc/effective_go.html

File: wordcount.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import "strings"
+
+// Optimization 1: used a fixed sized byte buffer to stream parts of a file in at a time
+// Optimization 2: don't convert the file content to an array of words, directly insert into the map of string to int
+func main() {
+  filename := os.Args[1]
+  fmt.Printf("%v given as filename arg\n", filename)
+
+  file, err := os.Open(filename)
+  defer file.Close()
```
Comment (kelmerp) [inline wordcount.go]: defer after checking for the error...in the case of an error there is probably nothing to close

File: wordcount.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  count, err2 := file.Read(filecontent)
+  if err2 != nil { panic(err2) }
+  fmt.Printf("read %v bytes\n", count)
+
+  //wordcount := map[string]int{"r": 3, "b": 2, "a": 3, "aslsdf": 33, "dude":4, "werd": 11 }
+  wordcount := countwords(string(filecontent))
+  printwordcount(wordcount)
+}
+
+func countwords(fc string) map[string]int {
```
Comment (kelmerp) [inline wordcount.go]: camelcase....so countWords

File: wordcount.go (hunk 3 above)
Comment (ramrom) [reply wordcount.go]: oh nice catch.

File: wordcount.go (hunk 2 above)
Comment (ramrom) [reply wordcount.go]: hehe, yea, i saw everyone elses code put these on new lines. 

even if it's a simple statement like this?

**Comment by magicalbanana** [comment]
I think I remember doing this challenge where I didn't use a struct but purely a `map[int]map[string]int` and did a sorting function on it as well. I think for different styles of data sets, you guys might want to consider adding that.

File: wordcount.go (hunk 3 above)
Comment (magicalbanana) [reply wordcount.go]: @ramrom haha. I'm someone still knows me. :D

**Pull request closed without merging**